internal/controller/util/mw_util.go

```diff
@@ -298,6 +364,28 @@ func Namespace(name string) *corev1.Namespace {
 	}
 }
 
+func ExtractResourceFromManifestWork(
+	mw *ocmworkv1.ManifestWork,
+	object client.Object,
+	gvk schema.GroupVersionKind,
+) error {
+	rawObject, err := GetRawExtension(mw.Spec.Workload.Manifests, gvk)
+	if err != nil {
+		return fmt.Errorf("failed fetching MaintenanceMode from manifest %w", err)
```

Comment:
failed fetching resource from manifest? Not MaintenanceMode

Reply:
Fixed